### ISSUE TITLE
Physical SIM got deactivated out of nowhere. Now i can not activate it again.

### ISSUE THREAD

**Issue by cybersa** [PixelExperience/android-issues#3514]
## Build date

PixelExperience_Plus_crosshatch-12.1-20220813-0635-OFFICIAL

## Expected Behavior

Able to activate Physical SIM

## Current Behavior

Nothing happens after i click on yes when activating the SIM

## Possible Solution

Reason for the issue: May be SIM settings got updated from Vi operator. Because issue started to happen out of nowhere.

## Steps to Reproduce

1. I got this notification out of nowhere:
![issue](https://user-images.githubusercontent.com/9217545/194887581-85db1625-d70a-4976-9f75-409d1ea81f1a.jpg)
2. After that my sim got deactivated, then i restarted my devices hoping that it will get activated again but no luck.
3. When i try to activate SIM manually from the Settings. Nothing happens:
![image](https://user-images.githubusercontent.com/9217545/194888015-16d1a428-3687-4a55-9a96-4a1ae13fc979.png)

Attached both logcat and radio logs.

/device crosshatch
/version twelve_plus

**Comment by pixelexperiencebot** [comment]
Issue created! You can close at any time by commenting ```/close```

**Comment by cybersa** [comment]
Attached logs:
[logcat.txt](https://github.com/PixelExperience/android-issues/files/9746757/logcat.txt)

**Comment by cybersa** [comment]
I think this issue can be resolved if we Reset settings through Settings -> System -> Reset Wi-FI,Mobile & Bluetooth.

What I did to resolve this issue?
1. Dirty flash PE 13 over PE 12
2. Still SIM was not activated. Tried to activate again no luck.
3. Reset Wi-FI,Mobile & Bluetooth through settings
4. Again tried to activate again no luck.
5. Repeated step 3 and 4 again and it started to work again.

**Comment by cybersa** [comment]
Today i faced the same issue again. But this time i forced to Wipe the data to make it work again. 
Attached the logcat again.
[logcat.txt](https://github.com/PixelExperience/android-issues/files/9912490/logcat.txt)

**Comment by MrPaul9NIX** [comment]
Same problem, at least every 6 days I have to force wipe data for it to work, I can't find any other solution

**Comment by cybersa** [comment]
Hello @MrPaul9NIX,
I have few questions for you:

1. Did you get this notification before the issue?
![Screenshot_20221101-181718_Nova7](https://user-images.githubusercontent.com/9217545/199418999-1765bbeb-5919-4829-93bc-a78b1058241c.png)
2. Are you facing this issue only on this Custom Rom? I didn't face this issue on Lineage but some times I faced auto disconnect and reconnect of Networks like Mobile , Wifi etc...
3. Are you using Magisk? I'm using Magisk and SafetyFix module. 
4. Are you on custom kernel? I'm using default kernel. 

Thanks.

**Comment by MrPaul9NIX** [comment]
Hi @cybersa 
Yes, it stopped recognizing the sim after that message, I have installed several roms and with no other it happens to me more than with this one, I do not have magisk and I am using the default kernel

**Comment by cybersa** [comment]
Thanks @MrPaul9NIX for the update.

**Comment by meyertime** [comment]
I am also facing this same issue.  It was working fine for a couple months since I first set up PixelExperience 12.  I also got that same notification.  I have a Pixel 3 (blueline).  I tried resetting wi-fi, mobile, bluetooth as mentioned, and no change.  I'm now backing up my data and going to try dirty flash to PE 13.  If that fails, then I guess wipe data is next, but I would really appreciate an option that doesn't involve that.

Some additional information: When I first set up PE, I used the option to set up the SIM later when going through the initial device setup.  I had this same issue then, but I wasn't sure if it was just my device or SIM card or whatever.  I finally did a factory reset and set up the SIM through the initial device setup, and it worked, and it has been working since then until I got that notification today.  Now it's behaving the same as it did before when it was broken.  So in other words, that may be a way to reproduce the issue: Skip SIM setup during initial device setup, then try to set up the SIM later on.

Also, I noticed that when there is no SIM card and I select "Add a network", I get a message that says "To use a new SIM card, you'll need to remove the one that's currently in the device".  This is wrong, because there's no SIM in the device.  My other Pixel 3 which has the stock ROM with no SIM card inserted instead says "If you have a SIM card, insert it now".  So it seems like whatever's broken is making the OS think there's a SIM card inserted when there isn't, and it could also be messing with it when there actually is a SIM card.  I'm thinking it could actually be related to #3463.  Like if it thinks there's an eSIM so it's ignoring the physical SIM, but the eSIM isn't getting deleted with the reset.  Or if it thinks the eSIM is actually a physical SIM so it doesn't delete it with the reset, and there's no way to remove it because it's not actually a physical SIM.

**Comment by ReallySnow** [comment]
> I am also facing this same issue. It was working fine for a couple months since I first set up PixelExperience 12. I also got that same notification. I have a Pixel 3 (blueline). I tried resetting wi-fi, mobile, bluetooth as mentioned, and no change. I'm now backing up my data and going to try dirty flash to PE 13. If that fails, then I guess wipe data is next, but I would really appreciate an option that doesn't involve that.
> 
> Some additional information: When I first set up PE, I used the option to set up the SIM later when going through the initial device setup. I had this same issue then, but I wasn't sure if it was just my device or SIM card or whatever. I finally did a factory reset and set up the SIM through the initial device setup, and it worked, and it has been working since then until I got that notification today. Now it's behaving the same as it did before when it was broken. So in other words, that may be a way to reproduce the issue: Skip SIM setup during initial device setup, then try to set up the SIM later on.
> 
> Also, I noticed that when there is no SIM card and I select "Add a network", I get a message that says "To use a new SIM card, you'll need to remove the one that's currently in the device". This is wrong, because there's no SIM in the device. My other Pixel 3 which has the stock ROM with no SIM card inserted instead says "If you have a SIM card, insert it now". So it seems like whatever's broken is making the OS think there's a SIM card inserted when there isn't, and it could also be messing with it when there actually is a SIM card. I'm thinking it could actually be related to #3463. Like if it thinks there's an eSIM so it's ignoring the physical SIM, but the eSIM isn't getting deleted with the reset. Or if it thinks the eSIM is actually a physical SIM so it doesn't delete it with the reset, and there's no way to remove it because it's not actually a physical SIM.

Thanks for the detailed explanation, I should now know what the problem is

**Comment by ReallySnow** [comment]
It is expected to be fixed in the next release

**Comment by pixelexperiencebot** [comment]
Issue closed by maintainer request.

**Comment by meyertime** [comment]
> It is expected to be fixed in the next release

That's great news! Any idea when the next release will be? If it will be a while, is there a way I can manually build the fix and try it? It would also help validate the fix ahead of release. Thanks!

In other news, the upgrade to PE 13 did not work. I still haven't done a full wipe, and I'm hoping to avoid it.

**Comment by ReallySnow** [comment]
> > It is expected to be fixed in the next release
> 
> That's great news! Any idea when the next release will be? If it will be a while, is there a way I can manually build the fix and try it? It would also help validate the fix ahead of release. Thanks!
> 
> In other news, the upgrade to PE 13 did not work. I still haven't done a full wipe, and I'm hoping to avoid it.

but I‘m p3 is Japanese version, and the Japanese version cannot use esim, so I hope someone can cooperate with me to test

**Comment by meyertime** [comment]
> but I‘m p3 is Japanese version, and the Japanese version cannot use esim, so I hope someone can cooperate with me to test

Not sure what you mean by p3, but I can help test.

I realized that my last message was not very clear. The upgrade to PE 13 worked, but it did not fix the problem with the SIM.

I'm using my old Pixel 3 with stock ROM as my daily driver now, so I can use my newer Pixel 3 to test PE.

**Comment by cmtsij** [comment]
@meyertime I saw the Novermber PE 13 plus version inlude the esim fix. you can check the release note.
Could you help to confirm the  PE 13 13 plus version fix the issue ?
Thanks

**Comment by meyertime** [comment]
@cmtsij @ReallySnow Long story short, the PE 13 Plus release worked!  Thanks for the quick turnaround!

I dirty flashed the PE 13 Plus 11/14 release.  After that, when I went into settings and "Add a network", it correctly said "If you have a SIM card, insert it now".  After inserting the SIM, I had to go back and tap "Activate Verizon" and "Yes", and then it actually did switch and start working.  Thanks again!
